kernel/hConstr.ml

```diff
@@ -267,7 +271,10 @@ let nonrel_leaf tbl c = match kind c with
     Tbl.raw_find tbl h (fun c' -> eq_leaf c c')
   | _ -> None
 
+let steps = ref 0
```

Comment:
Global references are not great, even for debugging purposes. Could you somehow store it directly in the local_env in of_constr?

Reply:
I guess we can put the tbl there while we're at it, it's not evolving any differently than the binder tables which are already there